packages/react/src/theme/theme.tsx

```diff
@@ -1,50 +1,56 @@
-import './styles/theme-dx-dark.scss';
-import './styles/variables-dark.scss';
-import './styles/variables-light.scss';
-import './styles/theme-dx-light.scss';
 import { currentTheme as currentVizTheme, refreshTheme } from 'devextreme/viz/themes';
 import React, { useCallback, useEffect, useMemo, useState } from 'react';
 
-const themeNames = ['dark', 'light'] as const;
-const themeStylesSheets = {};
-const storageKey = 'themeViewer';
+const themes = ['light', 'dark'] as const;
+const storageKey = 'app-theme';
+const themePrefix = 'app-theme-';
 
-export type Theme = typeof themeNames[number];
+const prefixes = ['./styles/theme-dx-', './styles/variables-'];
 
-function getCurrentTheme(): Theme {
-  return window.localStorage[storageKey] || 'light';
-}
+const loadStylesImports = async() => {
+  await Promise.all([
+    ...prefixes.flatMap((prefix) => [
+      import(/* webpackChunkName: "app-theme-dark" */ `${prefix}dark.scss`),
+      import(/* webpackChunkName: "app-theme-light" */ `${prefix}light.scss`)
+    ]),
+  ]);
+};
 
-function toArray<T>(arrayLikeObject) {
-  return [...arrayLikeObject as unknown as T[]];
-}
+export type Theme = typeof themes[number];
 
-function getThemeStyleSheets(): CSSStyleSheet[] {
-  const themePrefix = 'app-theme-';
+function getNextTheme(theme?: Theme) {
+  const index = !theme ? 0 : (themes.indexOf(theme) + 1);
+  return themes[
+    themes.length === index ? 0 : index
+  ];
+}
 
-  if (Object.keys(themeStylesSheets).length == themeNames.length) {
-    return Object.values<CSSStyleSheet>(themeStylesSheets).flat();
-  }
+function getCurrentTheme(): Theme {
+  return window.localStorage[storageKey] || getNextTheme();
+}
 
-  themeNames.forEach((themeName) => {
-    if (themeStylesSheets[themeName]) {
-      return;
-    }
+function isThemeStyleSheet(styleSheet, theme: Theme) {
+  const themeMarker = `${themePrefix}${theme}`;
+  // eslint-disable-next-line no-undef
+  return process.env.NODE_ENV === 'production' ?
```

Comment:
Too complicated logic. Could we simplify it? May be move it to method and add more variables

Reply:
simplified